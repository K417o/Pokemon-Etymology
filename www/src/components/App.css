/* layout */

body {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  max-width: 100vW;
  margin: 0 auto;
  padding-top: 15px;
  background: #001219;
}

.content {
  max-width: 1200px;
  margin: 0 auto;
}

.spacer {
  aspect-ratio: 900/450;
  width: 100%;
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
}

.layer1 {
  background-image: url('./images/layered-waves.svg');
}



#Cards {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}