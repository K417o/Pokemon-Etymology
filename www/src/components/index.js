import MultipleSelect from "./selectComponents/MultipleSelect";
import PkmnCard from "./PkmnCard/PkmnCard";
import NameTable from "./NameTable/NameTable";
import SingleSelect from "./selectComponents/SingleSelect";
import MultipleLabelKey from "./selectComponents/MultipleLabelKey";

export {MultipleSelect, SingleSelect, MultipleLabelKey, PkmnCard, NameTable};