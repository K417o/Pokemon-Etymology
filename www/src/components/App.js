import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'react-bootstrap-typeahead/css/Typeahead.css';
import { Container, Form } from 'react-bootstrap';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import './App.css';
import { MultipleSelect, MultipleLabelKey, SingleSelect, PkmnCard } from './index';
import { getAll } from '../services/selectData';
import { filterData } from '../services/filterData';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.typeSelector = React.createRef();
    this.colourSelector = React.createRef();
    this.pokemonSelector = React.createRef();
    this.categorySelector = React.createRef();
    this.originSelector = React.createRef();

    this.state = {
      types: [],
      categories: [],
      pokemon: [],
      colours: [],
      results: null
    }

    this.createCards = this.createCards.bind(this);
  }

  async getData() {
    let types = this.typeSelector.current;
    let colours = this.colourSelector.current;
    let pokemon = this.pokemonSelector.current;
    let categories = this.categorySelector.current;
    let strict = document.getElementById("checkbox").checked;
    let origins = this.originSelector.current;
    let result;
    if (pokemon.state.selected.length > 0) {
      result = [await filterData.getPkmn(pokemon.state.selected)];
    } else {
      result = await filterData.getResults(types.state.selected, strict, colours.state.selected, categories.state.selected, origins.state.selected);
    }
    this.setState({ results: result })
  };

  async componentDidMount() {
    this.setState({
      types: await getAll.types(),
      categories: await getAll.categories(),
      pokemon: await getAll.pokemon(),
      colours: await getAll.colours(),
      origins: await getAll.origins()
    })
  }

  createCards = () => {
    let cards = [];

    if (this.state.results != null) {
      let pkmn = this.state.results;
      for (let i in pkmn) {

        let click = true;
        cards.push(
          <PkmnCard
            name={pkmn[i].name}
            types={pkmn[i].types}
            height={pkmn[i].height}
            weight={pkmn[i].weight}
            genus={pkmn[i].genus}
            image={pkmn[i].image}
            colour={pkmn[i].colour}
            shape={!!pkmn[i].shape.depiction ? pkmn[i].shape.depiction : pkmn[i].shape}
            origins={pkmn[i].origins}
            rest={pkmn[i]}
            click={click}
            key={pkmn[i].name}

          />)
      }
    } else {
      cards.push(<div className="bright-Text"><br />No Results found.</div>)
    }
    return cards;
  }

  render() {


    return (
      <>
        <div className="content">

          <h1 className="bright-Text">Pokémon-Origins</h1>

          <br />
          <Container>
            <Row className="Selects">
              <Col><MultipleSelect ref={this.typeSelector} options={this.state.types} name={"Types"} />
              </Col>
              <Col md="auto">
                <Form.Check
                  className="bright-Text"
                  type={"checkbox"}
                  id={"checkbox"}
                  label={"and"}
                />
              </Col>
              <Col><MultipleSelect ref={this.colourSelector} options={this.state.colours} name={"Colours"} /></Col>
            </Row>
            <Row className="Selects">
              <Col><SingleSelect ref={this.pokemonSelector} options={this.state.pokemon} name={"Pokémon"} /></Col>
              <Col><SingleSelect ref={this.categorySelector} options={this.state.categories} name={"Category"} /></Col>
              <Col><MultipleLabelKey ref={this.originSelector} options={this.state.origins} name={"Origin"} /></Col>
            </Row>
          </Container>

          <br />


          <button
            className="glow-on-hover"
            type="button"
            onClick={this.getData.bind(this)}>
            Search for matches
          </button>
          

          <div id="Cards" className="flex-container">
            {this.createCards()}
          </div>


        </div>
        <div class="spacer layer1">
        </div>
      </>
    );
  }
}
export default App;


