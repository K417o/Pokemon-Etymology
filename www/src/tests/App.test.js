import React from 'react';
import userEvent from '@testing-library/user-event';
import { AverageConsumption, LoginPage, RoomDetailsPage, RoomOverview, Fallback } from '../components/';
import { screen, waitFor } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReactDom from 'react-dom';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom/extend-expect';

import App from '../components/App';
import { MultipleLabelKey, MultipleSelect, NameTable, PkmnCard, SingleSelect } from '../components/index';

let container;

beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

const nameTableDummie = { "deu": [{ "title": "Abrakadabra", "description": "https://de.wikipedia.org/wiki/Abrakadabra#", "positionInName": "0", "language": "https://iso639-3.sil.org/code/deu" }], "eng": [{ "title": "Abrakadabra", "description": "https://de.wikipedia.org/wiki/Abrakadabra#", "positionInName": "0", "language": "https://iso639-3.sil.org/code/eng" }], "jpn": [{ "title": "Edgar Cayce", "description": "https://de.wikipedia.org/wiki/Edgar_Cayce#", "positionInName": "0", "language": "https://iso639-3.sil.org/code/jpn" }], "fra": [{ "title": "Abrakadabra", "description": "https://de.wikipedia.org/wiki/Abrakadabra#", "positionInName": "0", "language": "https://iso639-3.sil.org/code/fra" }], "kor": [{ "title": "Aus dem Japanischen übernommen.", "description": null, "positionInName": "0", "language": "https://iso639-3.sil.org/code/kor" }], "zho": [{ "title": "Phonetische Übersetzung aus dem Japanischen.", "description": null, "positionInName": "0", "language": "https://iso639-3.sil.org/code/zho" }] };

test('SnapshotTest App', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('SnapshotTest MultipleSelect', () => {
  const tree = renderer.create(<MultipleSelect options={["Abra", "Kadabra", "Simsala"]} name={"Pokémon"} />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('SnapshotTest SingleSelect', () => {
  const tree = renderer.create(<SingleSelect options={["Abra", "Kadabra", "Simsala"]} name={"Pokémon"} />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('SnapshotTest MultipleLabelKey', () => {
  const tree = renderer.create(<MultipleLabelKey options={["Abra", "Kadabra", "Simsala"]} name={"Pokémon"} />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('SnapshotTest PkmnCard', () => {
  const tree = renderer.create(
    <PkmnCard
      name={"Abra"}
      types={[
        "Psychic"
      ]}
      height={"0.9"}
      weight={"19.5"}
      genus={"Psi Pokémon"}
      image={"https://www.pokewiki.de/images/thumb/4/44/Sugimori_063.png/250px-Sugimori_063.png"}
      colour={"http://dbpedia.org/resource/Brown"}
      shape={"https://www.pokewiki.de/images/thumb/0/07/Silhouette_6_HOME.png/24px-Silhouette_6_HOME.png"}
      click={true}
      key={"Abra"}
    />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('SnapshotTest NameTable', () => {
  const tree = renderer.create(<NameTable nameParts={nameTableDummie} />).toJSON()
  expect(tree).toMatchSnapshot()
})
