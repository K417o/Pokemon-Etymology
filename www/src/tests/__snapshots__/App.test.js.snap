// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapshotTest App 1`] = `
Array [
  <div
    className="content"
  >
    <h1
      className="bright-Text"
    >
      Pokémon-Origins
    </h1>
    <br />
    <div
      className="container"
    >
      <div
        className="Selects row"
      >
        <div
          className="col"
        >
          <div>
            <div
              className="rbt"
              style={
                Object {
                  "outline": "none",
                  "position": "relative",
                }
              }
              tabIndex={-1}
            >
              <div
                className="rbt-input-multi form-control rbt-input"
                onClick={[Function]}
                onFocus={[Function]}
                tabIndex={-1}
              >
                <div
                  className="rbt-input-wrapper"
                >
                  <div
                    style={
                      Object {
                        "display": "flex",
                        "flex": 1,
                        "height": "100%",
                        "position": "relative",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      aria-haspopup="listbox"
                      autoComplete="off"
                      className="rbt-input-main"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Choose one or more Pokémon-Types..."
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "border": 0,
                          "boxShadow": "none",
                          "cursor": "inherit",
                          "outline": "none",
                          "padding": 0,
                          "width": "100%",
                          "zIndex": 1,
                        }
                      }
                      type="text"
                      value=""
                    />
                    <input
                      aria-hidden={true}
                      className="rbt-input-hint"
                      readOnly={true}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "borderColor": "transparent",
                          "boxShadow": "none",
                          "color": "rgba(0, 0, 0, 0.54)",
                          "left": 0,
                          "pointerEvents": "none",
                          "position": "absolute",
                          "top": 0,
                          "width": "100%",
                        }
                      }
                      tabIndex={-1}
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-md-auto"
        >
          <div
            className="bright-Text form-check"
          >
            <input
              className="form-check-input"
              disabled={false}
              id="checkbox"
              type="checkbox"
            />
            <label
              className="form-check-label"
              htmlFor="checkbox"
              title=""
            >
              and
            </label>
          </div>
        </div>
        <div
          className="col"
        >
          <div>
            <div
              className="rbt"
              style={
                Object {
                  "outline": "none",
                  "position": "relative",
                }
              }
              tabIndex={-1}
            >
              <div
                className="rbt-input-multi form-control rbt-input"
                onClick={[Function]}
                onFocus={[Function]}
                tabIndex={-1}
              >
                <div
                  className="rbt-input-wrapper"
                >
                  <div
                    style={
                      Object {
                        "display": "flex",
                        "flex": 1,
                        "height": "100%",
                        "position": "relative",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      aria-haspopup="listbox"
                      autoComplete="off"
                      className="rbt-input-main"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Choose one or more Pokémon-Colours..."
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "border": 0,
                          "boxShadow": "none",
                          "cursor": "inherit",
                          "outline": "none",
                          "padding": 0,
                          "width": "100%",
                          "zIndex": 1,
                        }
                      }
                      type="text"
                      value=""
                    />
                    <input
                      aria-hidden={true}
                      className="rbt-input-hint"
                      readOnly={true}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "borderColor": "transparent",
                          "boxShadow": "none",
                          "color": "rgba(0, 0, 0, 0.54)",
                          "left": 0,
                          "pointerEvents": "none",
                          "position": "absolute",
                          "top": 0,
                          "width": "100%",
                        }
                      }
                      tabIndex={-1}
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="Selects row"
      >
        <div
          className="col"
        >
          <div>
            <div
              className="rbt"
              style={
                Object {
                  "outline": "none",
                  "position": "relative",
                }
              }
              tabIndex={-1}
            >
              <div
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "height": "100%",
                    "position": "relative",
                  }
                }
              >
                <input
                  aria-autocomplete="both"
                  aria-expanded={false}
                  aria-haspopup="listbox"
                  autoComplete="off"
                  className="rbt-input-main form-control rbt-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Choose one Pokémon"
                  role="combobox"
                  type="text"
                  value=""
                />
                <input
                  aria-hidden={true}
                  className="rbt-input-hint"
                  readOnly={true}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "transparent",
                      "boxShadow": "none",
                      "color": "rgba(0, 0, 0, 0.54)",
                      "left": 0,
                      "pointerEvents": "none",
                      "position": "absolute",
                      "top": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div>
            <div
              className="rbt"
              style={
                Object {
                  "outline": "none",
                  "position": "relative",
                }
              }
              tabIndex={-1}
            >
              <div
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "height": "100%",
                    "position": "relative",
                  }
                }
              >
                <input
                  aria-autocomplete="both"
                  aria-expanded={false}
                  aria-haspopup="listbox"
                  autoComplete="off"
                  className="rbt-input-main form-control rbt-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Choose one Category"
                  role="combobox"
                  type="text"
                  value=""
                />
                <input
                  aria-hidden={true}
                  className="rbt-input-hint"
                  readOnly={true}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderColor": "transparent",
                      "boxShadow": "none",
                      "color": "rgba(0, 0, 0, 0.54)",
                      "left": 0,
                      "pointerEvents": "none",
                      "position": "absolute",
                      "top": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div>
            <div
              className="rbt"
              style={
                Object {
                  "outline": "none",
                  "position": "relative",
                }
              }
              tabIndex={-1}
            >
              <div
                className="rbt-input-multi form-control rbt-input"
                onClick={[Function]}
                onFocus={[Function]}
                tabIndex={-1}
              >
                <div
                  className="rbt-input-wrapper"
                >
                  <div
                    style={
                      Object {
                        "display": "flex",
                        "flex": 1,
                        "height": "100%",
                        "position": "relative",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      aria-haspopup="listbox"
                      autoComplete="off"
                      className="rbt-input-main"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Choose one or more Pokémon-Origin..."
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "border": 0,
                          "boxShadow": "none",
                          "cursor": "inherit",
                          "outline": "none",
                          "padding": 0,
                          "width": "100%",
                          "zIndex": 1,
                        }
                      }
                      type="text"
                      value=""
                    />
                    <input
                      aria-hidden={true}
                      className="rbt-input-hint"
                      readOnly={true}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "borderColor": "transparent",
                          "boxShadow": "none",
                          "color": "rgba(0, 0, 0, 0.54)",
                          "left": 0,
                          "pointerEvents": "none",
                          "position": "absolute",
                          "top": 0,
                          "width": "100%",
                        }
                      }
                      tabIndex={-1}
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <br />
    <button
      className="glow-on-hover"
      onClick={[Function]}
      type="button"
    >
      Search for matches
    </button>
    <div
      className="flex-container"
      id="Cards"
    >
      <div
        className="bright-Text"
      >
        No Results found.
      </div>
    </div>
  </div>,
  <div
    class="spacer layer1"
  />,
]
`;

exports[`SnapshotTest MultipleLabelKey 1`] = `
<div>
  <div
    className="rbt"
    style={
      Object {
        "outline": "none",
        "position": "relative",
      }
    }
    tabIndex={-1}
  >
    <div
      className="rbt-input-multi form-control rbt-input"
      onClick={[Function]}
      onFocus={[Function]}
      tabIndex={-1}
    >
      <div
        className="rbt-input-wrapper"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "height": "100%",
              "position": "relative",
            }
          }
        >
          <input
            aria-autocomplete="list"
            aria-haspopup="listbox"
            autoComplete="off"
            className="rbt-input-main"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Choose one or more Pokémon-Pokémon..."
            style={
              Object {
                "backgroundColor": "transparent",
                "border": 0,
                "boxShadow": "none",
                "cursor": "inherit",
                "outline": "none",
                "padding": 0,
                "width": "100%",
                "zIndex": 1,
              }
            }
            type="text"
            value=""
          />
          <input
            aria-hidden={true}
            className="rbt-input-hint"
            readOnly={true}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "boxShadow": "none",
                "color": "rgba(0, 0, 0, 0.54)",
                "left": 0,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
            tabIndex={-1}
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SnapshotTest MultipleSelect 1`] = `
<div>
  <div
    className="rbt"
    style={
      Object {
        "outline": "none",
        "position": "relative",
      }
    }
    tabIndex={-1}
  >
    <div
      className="rbt-input-multi form-control rbt-input"
      onClick={[Function]}
      onFocus={[Function]}
      tabIndex={-1}
    >
      <div
        className="rbt-input-wrapper"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "height": "100%",
              "position": "relative",
            }
          }
        >
          <input
            aria-autocomplete="list"
            aria-haspopup="listbox"
            autoComplete="off"
            className="rbt-input-main"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Choose one or more Pokémon-Pokémon..."
            style={
              Object {
                "backgroundColor": "transparent",
                "border": 0,
                "boxShadow": "none",
                "cursor": "inherit",
                "outline": "none",
                "padding": 0,
                "width": "100%",
                "zIndex": 1,
              }
            }
            type="text"
            value=""
          />
          <input
            aria-hidden={true}
            className="rbt-input-hint"
            readOnly={true}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "transparent",
                "boxShadow": "none",
                "color": "rgba(0, 0, 0, 0.54)",
                "left": 0,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
            tabIndex={-1}
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SnapshotTest NameTable 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        Language
      </th>
      <th>
        Name and Explanation
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        German
      </td>
      <td>
        <span
          aria-describedby={null}
          className=""
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title={null}
        >
          Abrakadabra
        </span>
      </td>
    </tr>
    <tr>
      <td>
        English
      </td>
      <td>
        <span
          aria-describedby={null}
          className=""
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title={null}
        >
          Abrakadabra
        </span>
      </td>
    </tr>
    <tr>
      <td>
        Japanese
      </td>
      <td>
        <span
          aria-describedby={null}
          className=""
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title={null}
        >
          Edgar Cayce
        </span>
      </td>
    </tr>
    <tr>
      <td>
        French
      </td>
      <td>
        <span
          aria-describedby={null}
          className=""
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title={null}
        >
          Abrakadabra
        </span>
      </td>
    </tr>
    <tr>
      <td>
        Korean
      </td>
      <td>
        <span>
          Aus dem Japanischen übernommen.
        </span>
      </td>
    </tr>
    <tr>
      <td>
        Chinese
      </td>
      <td>
        <span>
          Phonetische Übersetzung aus dem Japanischen.
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`SnapshotTest PkmnCard 1`] = `
<div
  aria-controls="example-collapse-text-Abra"
  aria-expanded={false}
  className="pkmn-card clickable_true"
  onClick={[Function]}
>
  <div
    className="pkmn-card-container border-gray rounded border mx-2 my-3 d-flex flex-row align-items-center p-0 bg-light"
  >
    <div
      className="container"
    >
      <div
        className="main-Info-row row"
      >
        <div
          className="col-md-auto"
        >
          <img
            alt="Abra Artwork"
            className="d-block original-pic"
            src="https://www.pokewiki.de/images/thumb/4/44/Sugimori_063.png/250px-Sugimori_063.png"
          />
        </div>
        <div
          className="col"
        >
          <div
            className="px-3 details"
          >
            <h4>
              Abra
            </h4>
            <span
              className="country-region text-secondary"
            >
              Psi Pokémon
            </span>
            <br />
            <span
              className="country-region text-secondary "
            >
              Height: 0.9m
            </span>
            	
            <span
              className="country-region text-secondary "
            >
              Weight: 19.5kg
            </span>
          </div>
        </div>
        <div
          className="pkmn-icons col-md-auto"
        >
          <div>
            <img
              alt="Psychic Icon"
              className="type"
              src="Pokémon_Psychic_Type_Icon.svg"
            />
          </div>
          <div>
            <img
              alt="Abra-shape"
              className="shape"
              src="https://www.pokewiki.de/images/thumb/0/07/Silhouette_6_HOME.png/24px-Silhouette_6_HOME.png"
            />
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          aria-expanded={null}
          className="collapse"
        >
          <table
            className="table table-striped table-bordered table-hover"
          >
            <thead>
              <tr>
                <th>
                  Language
                </th>
                <th>
                  Name and Explanation
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
          <div
            className="card-group"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SnapshotTest SingleSelect 1`] = `
<div>
  <div
    className="rbt"
    style={
      Object {
        "outline": "none",
        "position": "relative",
      }
    }
    tabIndex={-1}
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "height": "100%",
          "position": "relative",
        }
      }
    >
      <input
        aria-autocomplete="both"
        aria-expanded={false}
        aria-haspopup="listbox"
        autoComplete="off"
        className="rbt-input-main form-control rbt-input"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Choose one Pokémon"
        role="combobox"
        type="text"
        value=""
      />
      <input
        aria-hidden={true}
        className="rbt-input-hint"
        readOnly={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "boxShadow": "none",
            "color": "rgba(0, 0, 0, 0.54)",
            "left": 0,
            "pointerEvents": "none",
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
        tabIndex={-1}
        value=""
      />
    </div>
  </div>
</div>
`;
